substrate/utils/frame/benchmarking-cli/src/pallet/writer.rs

```diff
@@ -175,6 +179,121 @@ fn map_results(
 	Ok(all_benchmarks)
 }
 
+// Calculates the total maximum weight of an extrinsic (if present, based on the max. component
+// value) and compares it with the max. extrinsic weight allowed in a single block.
+//
+// `max_extrinsic_weight` & `db_weight` are obtained from the runtime configuration.
+pub(crate) fn sanity_weight_check(
+	all_results: HashMap<(String, String), Vec<BenchmarkData>>,
+	max_extrinsic_weight: Weight,
+	db_weight: RuntimeDbWeight,
+	sanity_weight_check: SanityWeightCheck,
+) -> Result<(), std::io::Error> {
+	if sanity_weight_check == SanityWeightCheck::Ignore {
+		return Ok(())
+	}
+	// Helper function to return max. component value (i.e. max. complexity parameter).
+	fn max_component(parameter: &ComponentSlope, component_ranges: &Vec<ComponentRange>) -> u64 {
+		for component in component_ranges {
+			if parameter.name == component.name {
+				return component.max.into()
+			}
+		}
+		0
+	}
+
+	color_print::cprintln!(
```

Comment:
Should just do `log::warn!` as we cannot assume that the output terminal actually supports colors.